test/export/test_torchbind.py

```diff
@@ -16,6 +16,47 @@
 
 @skipIfTorchDynamo("torchbind not supported with dynamo yet")
 class TestExportTorchbind(TestCase):
+    def setUp(self):
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_Foo")
+        class FakeFoo:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+            @classmethod
+            def from_concrete(cls, foo):
+                (x, y), _ = foo.__getstate__()
+                return cls(x, y)
+
+            def add_tensor(self, z):
+                return (self.x + self.y) * z
+
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_TensorQueue")
+        class FakeTensorQueue:
+            def __init__(self, q):
+                self.queue = q
+
+            @classmethod
+            def from_concrete(cls, real_tq):
+                return cls(real_tq.clone_queue())
```

Comment:
I expected this to create a FakeTensorQueue with FakeTensors. Instead, it looks like we're cloning real Tensors and putting them into the FakeTensorQueue. What's up?

Reply:
Yeah, you're right. This example didn't fakify the internal states of FakeTensorQueue, and we should do it in this PR. I can think of two possible approaches of doing it: 

1. **We do the fakification.** Specifically, users deal with real tensors and use whatever ways they have/need to retrieve the state from the real tensorQueue and create a FakeTensorQueue instance.  We fakify the FakeTensorQueue further by recursively applying something like the wrap_fake of make_fx's to the FakeTensorQueue's attributes.

2. **Users do the fakification.** Users need to use the functions that we have in wrap_fake (or we could provide some user-facing fakification API) to create the FakeTensorQueue and we'll use the FakeTensoQueue as it is.

I'm more in favor of 1. because users may not always write the correct fakification logic. For example, forget to use the fakification API at all, doesn't fakify some tensor/integer, or doesn't recursively fakify the ScriptObject nested in current one. One example is an `add_tensor` method that returns `self.real + x`, where `x` is fake but `self.real` is not properly fakified).

Reply:
Resolved online. decided to use Option 2 as it's more explicit and has a clear contract.